### ISSUE TITLE
Accessibility bug: website home page link to home does not have an Accessible Name

### ISSUE THREAD

**Issue by ashleemboyer** [tailwindlabs/headlessui#2827]
On [the Headless UI home page](https://headlessui.com/), the home page link around the site logo in the top-level `<header>` element does not have an Accessible Name. This issue falls under two Level A WCAG 2.1 success criteria:

- [2.4.4 Link Purpose (In Context)](https://www.w3.org/TR/WCAG21/#link-purpose-in-context)
- [4.1.2 Name, Role, Value](https://www.w3.org/TR/WCAG21/#name-role-value)

**Comment by reinink** [comment]
Hey thanks for this suggestion, I've just added an `aria-label` to the logo link 👍

<img width="584" alt="image" src="https://github.com/tailwindlabs/headlessui/assets/882133/eaf6b14f-8c7a-41be-bb60-93971013cd04">
